java/code/src/com/suse/manager/reactor/messaging/RegistrationUtils.java

```diff
@@ -136,11 +141,15 @@ public static void finishRegistration(MinionServer minion, Optional<ActivationKe
             // SSH Minions need this to set last booted value.
             statesToApply.add(ApplyStatesEventMessage.SYSTEM_INFO);
         }
+
+        handleActivationKeyAppStreams(activationKey, statesToApply, statesToApplyPillar);
```

Comment:
I think this is called at the wrong place.
The `statesToApply` is a list which is executed in the order of the list.
With this order you might install packages from a stream you have not enabled yet.